http/vibe/http/websockets.d

```diff
@@ -567,8 +680,18 @@ final class WebSocket {
 	*/
 	void close(short code = 0, scope const(char)[] reason = "")
 	{
+		//assume the default is normal, intentional closure
+		if(code == 0)
+			code = WebSocketCloseReason.normalClosure;
```

Comment:
This would remove the possibility to send a closing frame *without*  a code/reason, which may be desirable for efficiency if a closing reason isn't used anyway. Do you think that sending no status is an issue for existing endpoints in practice? In that case, the other option would be to change the default parameter to `code = WebSocketCloseReason.normalClosure` and to define `WebSocketCloseReason.none = 0` to allow explicitly skipping the status.

Reply:
I like the idea of adding a `WebSocketCloseReason.none` and defaulting to normal.  I'm not sure if sending no status is an issue, though it probably acts as an unexpected closure.  From an API standpoint, I had expected that the default was a normal closure.

Reply:
My expectation was that it is usually trated as a normal closure, except if the peer application actually expects a status code, but I would be fine with the `.none` solution.

BTW, a related question is if the reason string should be omitted if none is given explicitly. Sending a generic string doesn't add much value and may be unwanted use of bandwidth for applications where this is critical.

Reply:
With the generic string, are you referring to the `0` case, or the spec-recognized cases, or both?  I admit that I like the idea of providing that string, though saving bandwidth is appealing too.

My motivations on this are driven by having to look up the close reason codes on the server when we receive a websocket close and also on the browser's side when we get a number only and I have to look it up.

Do you think that adding another method would be better, or maybe using the defaulting string behavior only for non-release versions?

Reply:
Please let me know how you want to go with this since everything else is green on the PR.

Reply:
As for the string, I think it would be fine to just treat `null` as a special case to still allow people to disable the reason phrase if bandwidth is critical. So the default phrase would just be set `if (reason !is null && reason.length == 0)`. Using different behavior for debug/release is something I'd rather not do in this case, since I'd rate the reason phrase to be slightly more than just a debugging aid (as opposed to sending a stack trace or similar things).